src/screens/ChannelListScreenV2/IncognitoEmptyChat/index.tsx

```diff
@@ -0,0 +1,151 @@
+import IconArrowRight from '../../../assets/icons/Ic_arrow_right';
+import dimen from '../../../utils/dimen';
+import {COLORS} from '../../../utils/theme';
+import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {ANONYMOUS} from '../../../hooks/core/constant';
+import useAnonymousChannelListScreenHook from '../../../hooks/screen/useAnonymousChannelListHook';
+import {DISCOVERY_TAB_TOPICS, NavigationConstants} from '../../../utils/constants';
+
+const {width: widthScreen} = Dimensions.get('window');
+
+const MODE_FULL = 1;
+const MODE_BUTTON = 2;
+const MODE_HIDE = 3;
+
+const IncognitoEmptyChat = ({totalChannel}) => {
+  const {goToContactScreen} = useAnonymousChannelListScreenHook();
+  const navigation = useNavigation();
+
+  const getMode = () => {
+    let result = MODE_FULL;
+    if (totalChannel === 0) {
+      result = MODE_FULL;
+    }
+    if (totalChannel >= 3) {
+      result = MODE_HIDE;
+    }
+    if ((totalChannel > 0 && totalChannel < 3) || widthScreen < 380) {
+      result = MODE_BUTTON;
+    }
+
+    return result;
+  };
+
+  const IncognitoButton = ({title, subtitle, onPress}) => (
+    <TouchableOpacity style={styles.item} onPress={onPress}>
+      <View style={styles.itemContent}>
+        <Text style={styles.itemTitle}>{title}</Text>
+        <Text style={styles.itemSubtitle}>{subtitle}</Text>
+      </View>
+      <View>
+        <IconArrowRight width={8} height={12} fill={COLORS.gray400} />
+      </View>
+    </TouchableOpacity>
+  );
+
+  if (getMode() === MODE_HIDE) return null;
+
+  return (
+    <View style={styles.container} testID="IncognitoEmptyChatContainer">
+      {getMode() === MODE_FULL && (
+        <Image testID="IncognitoEmptyChatImage" source={EmptyIncognito} style={styles.image} />
+      )}
+      {getMode() === MODE_FULL && (
+        <Text testID="IncognitoEmptyChatText" style={styles.titleText}>
+          All your incognito activity, including chats, posts, comments and communities will appear
+          here.
+        </Text>
+      )}
+      <View style={styles.wrapper}>
+        {getMode() === MODE_FULL && <View testID="IncognitoEmptyChatLine" style={styles.line} />}
+        <IncognitoButton
+          title="Start an Incognito Chat"
+          subtitle="Send an incognito message to friends."
+          onPress={() => goToContactScreen({from: ANONYMOUS})}
+        />
+        <IncognitoButton
+          title="Create an Incognito Post"
+          subtitle="Send an incognito message to friends."
+          onPress={() =>
+            navigation.navigate(NavigationConstants.CREATE_POST_SCREEN, {followType: 'incognito'})
+          }
+        />
+        <IncognitoButton
+          title="Join Communities in Incognito Mode"
+          subtitle="Safely join without anyone knowing."
+          onPress={() => navigation.navigate('DiscoveryScreen', {tab: DISCOVERY_TAB_TOPICS})}
+        />
+      </View>
+    </View>
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The repeated calls to `getMode()` should be replaced with a single call stored in a variable as mentioned earlier. This will improve performance by reducing unnecessary function calls.



<!-- This is an auto-generated comment by OSS CodeRabbit -->